lib/unsplash/photo.rb

```diff
@@ -57,10 +59,18 @@ def random(categories: nil, collections: nil, featured: nil, user: nil, query: n
           h:        height,
           orientation: orientation
         }.select { |k,v| v }
-
-        photo = Unsplash::Photo.new JSON.parse(connection.get("/photos/random", params).body)
-        photo.user = Unsplash::User.new photo.user
-        photo
+        if count
+          params[:count] = count
+          photos = parse_list connection.get("/photos/random/", params).body
```

Comment:
I'd recommend against using `parse_list` here in combination with your `photos.map` on line 65.  `parse_list` is already looping through the results, so maybe it would make more sense to have it accept a block that you call to override the default building of the photo object?  Something like:

```ruby
def parse_list(json)
  if block_given?
    JSON.parse(json).map { |photo| yield(photo) }
  else
    JSON.parse(json).map { |photo| new photo }
  end
end
```

Then you can update the above line to:
```ruby
photos = parse_list(connection.get("/photos/random/", params).body) do |photo_params|
  photo = new(photo_params)
  photo.user = Unsplash::User.new photo[:user]
  photo
end
```